### ISSUE TITLE
Engineering format for large numbers

### ISSUE THREAD

**Issue by krlmlr** [r-lib/pillar#41]
https://en.wikipedia.org/wiki/Engineering_notation

CC @christophsax

**Comment by hadley** [comment]
I think we should consider this the default for large numbers

**Comment by randomgambit** [comment]
this package is clearly going in the right direction!! :)

**Comment by krlmlr** [comment]
This is now implemented, at most 13 decimal places (including the dot) are printed.

``` r
library(tibble)
tibble(x = 1e12)
#> # A tibble: 1 x 1
#>               x
#>           <dbl>
#> 1 1000000000000
tibble(x = 1e13)
#> # A tibble: 1 x 1
#>         x
#>     <dbl>
#> 1 1.00e13
tibble(x = 1e-10)
#> # A tibble: 1 x 1
#>              x
#>          <dbl>
#> 1 0.0000000001
tibble(x = 1e-11)
#> # A tibble: 1 x 1
#>          x
#>      <dbl>
#> 1 1.00e-11
```

<sup>Created on 2019-05-03 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by hadley** [comment]
The exponents aren't divisible by three in your examples

**Comment by krlmlr** [comment]
Right, I was confused.

**Comment by randomgambit** [comment]
sorry why would they need to be divisible by three?!

**Comment by krlmlr** [comment]
That's what makes engineering notation different from scientific notation.

**Comment by krlmlr** [comment]
Would you like to test-drive `num()`, available in pillar 1.5.1.9002, to be released as pillar 1.6.0? After release, I'll reexport this the constructor from tibble too. See https://pillar.r-lib.org/dev/articles/numbers.html for a documentation stub.

``` r
library(pillar)
library(tibble)

x <- c(543123578890.23, 240234.131, 40234.1)
col_a <- num(x, notation = "eng")

col_a
#> <tibble_num(eng)[3]>
#> [1] 543.00e9 240.00e3  40.20e3
tibble::tibble(x, col_a)
#> # A tibble: 3 x 2
#>               x    col_a
#>           <dbl>    <eng>
#> 1 543123578890. 543.00e9
#> 2       240234. 240.00e3
#> 3        40234.  40.20e3
```

<sup>Created on 2021-04-03 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

The scientific display is still a bit off, working on it.
